### PR TITLE
[KOTLIN Spring] add interfaceOnly option

### PR DESCRIPTION
**Pull request by fj-roman**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@jimschubert (2017/09) heart, @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04)

### Description of the PR

Add interfaceOnly option for kotlin-spring. Now you can write down you own Controller Classes and implement the API-Interfaces.
I'm not certain about the combination of the properties interfaceOnly and serviceImplementation. Maybe it's confusing.

### CODE REVIEW

**Comment by wing328** [comment]
Can you please resolve the merge conflicts when you've time?

File: modules/openapi-generator/src/main/resources/kotlin-spring/api.mustache [hunk 1]
```diff
@@ -57,19 +55,19 @@ import kotlin.collections.Map
 @RequestMapping("\${api.base-path:<%contextPath%>}")
 <%={{ }}=%>
 {{#operations}}
-class {{classname}}Controller({{#serviceInterface}}@Autowired(required = true) val service: {{classname}}Service{{/serviceInterface}}) {
+interface {{classname}} {
```
Comment (jimschubert) [inline modules/openapi-generator/src/main/resources/kotlin-spring/api.mustache]: Changing `api.mustache` to an interface class breaks convention we use elsewhere, and will cause anyone with custom templates to break (as the default `interfaceOnly=false` loads a file that didn't previously exist.

**Comment by jimschubert** [comment]
I've tagged this as "Breaking change without fallback" since it will break custom templates as-is.
Let me know if you'd like assistance in removing the breaking change and I can open a PR against this branch in your fork.

**Comment by fj-roman** [comment]
@jimschubert @wing328 I think the Shippable check is broke at the moment. I don't see any mistake by the PR.

**Comment by jimschubert** [comment]
@fj-roman shippable has been failing to download dependencies for the docker image today.

**Comment by fj-roman** [comment]
@jimschubert I removed the breaking change can you review the PR again please.

**Comment by jimschubert** [comment]
The breaking change was removed. This looks good at quick glance, but I'll try it out tonight and merge if no other concerns.

**Comment by jimschubert** [comment]
@fj-roman sorry it took a bit to evaluate after you pinged me on Gitter, but this has been merged as of a couple days ago. Just wanted to notify in case you didn't see.  Thanks for the contribution!

**Comment by DannyRussellWP** [comment]
@jimschubert Just an FYI that the readme for kotlin spring generator doesn't include this option.

**Pull request merged**